### PR TITLE
feat(scully) CLI option to write scanRoutes Statistics to scullyStats.json

### PR DESCRIPTION
**Pull request by kananlanginhooper**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ x ] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ x ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

Build Statistics

npx scully --stats

Save the latest build statistics to a json file. Information includes the number of routes, as well as various times (all represented in seconds). The file is named scullyStats.json, and is overwritten with each build. Json file layout is as such:

{
  "numberOfRoutes": 51,
  "generatingTime": 55.63,
  "routesPerSecond": 0.92,
  "findingRoutesAngular": 2.71838,
  "routeDiscovery": 0.00026,
  "renderingPages": 52.21755
}

Issue Number: N/A

## What is the new behavior?

 no change unless the --stats cli flag is used

## Does this PR introduce a breaking change?

- [ ] Yes
- [ x ] No

### CODE REVIEW

File: libs/scully/src/lib/utils/handlers/defaultAction.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ async function plugin(localBaseFilter = baseFilter): Promise<HandledRoute[]> {
     const processedRoutes = await findPlugin(processRoutes)(handledRoutes);
 
     /** handleRouteDiscoveryDone run the discoverydone plugins */
-    const discoveryDone = handleRouteDiscoveryDone(processedRoutes);
+    const discoveryDone = await handleRouteDiscoveryDone(processedRoutes);
```
Comment (SanderElias) [inline libs/scully/src/lib/utils/handlers/defaultAction.ts:43]: This await must not be there. We don't need to `await` for this to complete.
if you want to reuse the results use `processedRoutes` plugins.

File: libs/scully/src/lib/utils/handlers/handleRouteDiscoveryDone.ts [hunk 2]
```diff
@@ -9,6 +9,16 @@ export async function handleRouteDiscoveryDone(handledRoutes: HandledRoute[]) {
...
+  const plugins_routeDiscoveryDone = plugins.routeDiscoveryDone;
+
+  const plugins_routeDiscoveryDone_values = Object.values(plugins_routeDiscoveryDone);
+
+  const arrPlugins = plugins_routeDiscoveryDone_values.map((plugin) => {
+    return plugin(clone);
+  });
+
+  await Promise.all(arrPlugins);
+
```
Comment (SanderElias) [inline libs/scully/src/lib/utils/handlers/handleRouteDiscoveryDone.ts:22]: For the same reason, this code is unneeded

File: scully.sample-blog.config.ts [hunk 3]
```diff
@@ -157,6 +157,10 @@ export const config: Promise<ScullyConfig> = (async () => {
     guessParserOptions: {
       excludedFiles: ['apps/sample-blog/src/app/exclude/exclude-routing.module.ts'],
     },
+    puppeteerLaunchOptions: {
+      headless: true,
+      executablePath: 'C:/Program Files (x86)/Google/Chrome/Application/chrome.exe',
+    },
```
Comment (SanderElias) [inline scully.sample-blog.config.ts:163]: This is very depeding on how your system is setup. I would rather not have this in.

File: scully.scully-docs.config.ts [hunk 4]
```diff
@@ -63,8 +65,8 @@ export const config: ScullyConfig = {
     },
   },
   puppeteerLaunchOptions: {
-    defaultViewport: null,
-    devtools: false,
+    headless: true,
+    executablePath: 'C:/Program Files (x86)/Google/Chrome/Application/chrome.exe',
```
Comment (SanderElias) [inline scully.scully-docs.config.ts:69]: Same here

**Review by SanderElias — COMMENTED** [review]
Perhaps we should do a quick meet, so you can update me on what it is your trying to accomplish?

File: libs/scully/src/lib/utils/handlers/defaultAction.ts (hunk 1 above)
Comment (kananlanginhooper) [reply libs/scully/src/lib/utils/handlers/defaultAction.ts]: Sorry about that, didn't mean for that to be committed.

File: libs/scully/src/lib/utils/handlers/handleRouteDiscoveryDone.ts (hunk 2 above)
Comment (kananlanginhooper) [reply libs/scully/src/lib/utils/handlers/handleRouteDiscoveryDone.ts]: This was for my internal test, just so I could be sure what was happening here.  Didn't mean for that to be committed.

File: scully.sample-blog.config.ts (hunk 3 above)
Comment (kananlanginhooper) [reply scully.sample-blog.config.ts]: Wow, really sorry that got left in there.

File: scully.scully-docs.config.ts (hunk 4 above)
Comment (kananlanginhooper) [reply scully.scully-docs.config.ts]: again, sorry about that.

**Review by kananlanginhooper — COMMENTED** [review]
Should all be cleaned up there.

**Review by SanderElias — APPROVED** [review]

**Pull request merged**